### PR TITLE
Uplink Revamp

### PR DESCRIPTION
**Pull request by Geevies**
Requires: https://github.com/Aurorastation/Aurora.3/pull/7996

* Added a bunch of descriptions to the traitor uplink so you can be more sure of what you're buying.
* Added various new weapons, tools, and kits to the traitor uplink.
* Various traitor items have had their cost tweaked to bring them into a tighter balance.
* The Mecha dropper has been removed from the revolution uplink and added to the general uplink under devices and tools.
* The Tommygun is now know as the Vintage Sub-Machine gun everywhere.
* Manhacks are now called Viscerators everywhere.

Feedback Thread:
https://forums.aurorastation.org/topic/13609-feedbackhype-uplink-revamp/

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/datums/uplink/ammunition.dm [hunk 1]
```diff
@@ -6,31 +6,61 @@
...
+	name = "14.5mm box"
+	item_cost = 8
 	path = /obj/item/storage/box/sniperammo
+	desc = "Contains seven rounds of 14.5mm bullets, for use with the anti-materiel rifle."
+
+/datum/uplink_item/item/ammo/shotgun_shells
+	name = "Box of Shells"
+	item_cost = 5
+	path = /obj/item/storage/box/shotgunshells
+	desc = "Contains eight shotgun shells."
```
Comment (NonQueueingMatt) [inline code/datums/uplink/ammunition.dm]: should specify that it's buckshot

File: code/datums/uplink/highly visible and dangerous weapons.dm [hunk 2]
```diff
@@ -109,12 +109,27 @@
 	item_cost = 2
 	path = /obj/item/material/star
 
+/datum/uplink_item/item/visible_weapons/plasma_shotty
+	name = "Plasma Shotgun"
```
Comment (NonQueueingMatt) [inline code/datums/uplink/highly visible and dangerous weapons.dm]: Probably a bit too good for ten?

**Comment by Kaedwuff** [comment]
4 telecrystals for a personal AI device is grossly overpriced. The item isn't even illegal, can easily be printed out on station by science, and it can even be brought on station through the loadouts.

I can't see you justifying charging more than 1-2 for this,

File: code/datums/uplink/devices and tools.dm [hunk 3]
```diff
@@ -138,3 +156,37 @@
...
 
+/datum/uplink_item/item/tools/keypad
+	name = "Keypad Mag-Lock"
+	item_cost = 4
+	path = /obj/item/device/magnetic_lock/keypad
+	desc = "A maglock that requires the user to enter a passcode to lock and then later unlock."
+
+/datum/uplink_item/item/tools/personal_ai
+	name = "Personal AI"
+	item_cost = 4
```
Comment (Geevies) [inline code/datums/uplink/devices and tools.dm]:
```suggestion
	item_cost = 2
```

File: code/datums/uplink/ammunition.dm [hunk 4]
```diff
@@ -6,31 +6,61 @@
...
-/datum/uplink_item/item/ammo/tommygunmag
-	name = "Tommygun Magazine (.45)"
+/datum/uplink_item/item/ammo/fourty_five
+	name = ".45 Pistol Magazine"
 	item_cost = 2
+	path = /obj/item/ammo_magazine/c45m
+	desc = "Contains seven rounds of .45 bullets."
+
+/datum/uplink_item/item/ammo/tommygundrum
+	name = "Sub-Machine Gun Drum Magazine (.45)"
```
Comment (Aticius) [inline code/datums/uplink/ammunition.dm]: Submachine is one word.

**Comment by Geevies** [comment]
i have the big dumb

File: code/datums/uplink/ammunition.dm [hunk 5]
```diff
@@ -6,31 +6,61 @@
...
 
-/datum/uplink_item/item/ammo/tommygunmag
-	name = "Tommygun Magazine (.45)"
+/datum/uplink_item/item/ammo/fourty_five
+	name = ".45 Pistol Magazine"
 	item_cost = 2
+	path = /obj/item/ammo_magazine/c45m
+	desc = "Contains seven rounds of .45 bullets."
+
+/datum/uplink_item/item/ammo/tommygundrum
```
Comment (Arrow768) [inline code/datums/uplink/ammunition.dm]: You should repath that

File: code/datums/uplink/ammunition.dm [hunk 6]
```diff
@@ -6,31 +6,61 @@
...
+	name = "Submachine Gun Drum Magazine (.45)"
+	item_cost = 6
+	path = /obj/item/ammo_magazine/tommydrum
```
Comment (Arrow768) [inline code/datums/uplink/ammunition.dm]: same here

File: code/datums/uplink/ammunition.dm [hunk 7]
```diff
@@ -6,31 +6,61 @@
...
+	desc = "Contains fifty rounds of .45 bullets, for use with the vintage submachine gun."
+
+/datum/uplink_item/item/ammo/tommygunmag
```
Comment (Arrow768) [inline code/datums/uplink/ammunition.dm]: same here

File: code/datums/uplink/ammunition.dm [hunk 8]
```diff
@@ -6,31 +6,61 @@
...
+	name = "Submachine Gun Magazine (.45)"
+	item_cost = 4
 	path = /obj/item/ammo_magazine/tommymag
```
Comment (Arrow768) [inline code/datums/uplink/ammunition.dm]: same here

File: maps/templates/orbital/orbital_templates.dm [hunk 9]
```diff
@@ -1,11 +1,23 @@
 /datum/map_template/mecha
-	name = "Mecha Drop"
+	name = "Combat Mecha Drop"
 	mappath = 'maps/templates/orbital/combat-mecha.dmm'
 
+/datum/map_template/mecha/heavy
+	name = "Heavy Mecha Drop"
+	mappath = 'maps/templates/orbital/heavy-mecha.dmm'
+
+/datum/map_template/mecha/light
```
Comment (Arrow768) [inline maps/templates/orbital/orbital_templates.dm]: My preference would be to make the light the default and then go from there with subpaths

File: code/datums/uplink/grenades.dm [hunk 10]
```diff
@@ -5,27 +5,36 @@
...
+	desc = "A box of five grenades that cause a risky EMP explosion, capable of toggling headsets off, permanently destroying IPC units and draining a stationbound completely."
 	
 /datum/uplink_item/item/grenades/frag
 	name = "5xFrag Grenades"
 	item_cost = 12
 	path = /obj/item/storage/box/frags
 
+/datum/uplink_item/item/grenades/cardox
+	name = "5xCardox Grenades"
+	item_cost = 10
```
Comment (Arrow768) [inline code/datums/uplink/grenades.dm]: Seems relative expensive compared to the EMP/Viscerator grenades.

File: code/datums/uplink/devices and tools.dm [hunk 11]
```diff
@@ -138,3 +150,37 @@
...
+	name = "Personal AI"
+	item_cost = 2
+	path = /obj/item/device/paicard
+	desc = "An unmodified personal AI that can assist you in your ventures."
+
+/datum/uplink_item/item/tools/pin_extractor
+	name = "Firing Pin Extractor"
+	item_cost = 2
+	path = /obj/item/device/pin_extractor
+	desc = "An extractor tool capable of extracting firing pins from all firearms, ever."
```
Comment (Arrow768) [inline code/datums/uplink/devices and tools.dm]:
```suggestion
	desc = "An extractor tool capable of extracting firing pins from most firearms."
```

File: code/datums/uplink/devices and tools.dm [hunk 12]
```diff
@@ -138,3 +150,37 @@
...
+
+/datum/uplink_item/item/tools/radio_jammer
+	name = "Radio Jammer"
+	item_cost = 5
+	path = /obj/item/device/radiojammer
+	desc = "A small jammer that can fit inside a pocket. Capable of disrupting nearby radios and hivenet transmitters."
+
+/datum/uplink_item/item/tools/jetpack
+	name = "Jetpack"
+	item_cost = 8
```
Comment (Arrow768) [inline code/datums/uplink/devices and tools.dm]: Seems relatively expensive for a utility

File: code/datums/uplink/grenades.dm (hunk 10 above)
Comment (Geevies) [reply code/datums/uplink/grenades.dm]: Cardox grenades are pretty toxic to all people except synths. I'll drop it down to 8.

**Pull request merged**